lib/vanagon/project.rb

```diff
@@ -476,5 +476,33 @@ def generate_package
     def generate_packaging_artifacts(workdir)
       @platform.generate_packaging_artifacts(workdir, @name, binding)
     end
+
+    # Generate a json file which lists all of the dependant components of the project
+    #
+    # @return [Hash] where the top level keys are components and their values
+    #   are hashes with additional information on the component.
+    def generate_dependencies_info
+      all_components = {}
+      @components.each do |component|
+        all_components = all_components.merge(component.get_dependency_hash)
+      end
+      all_components
+    end
+
+    # Writes a json file at `ext/build_metadata.json` containing information
+    # about what went into a built artifact
+    def generate_manifest_json
+      manifest = {
+        "packaging_type" => {
+          "vanagon" => Gem.loaded_specs["vanagon"].version.to_s,
+        },
+        "git_sha" => version,
+        "components" => generate_dependencies_info,
+        "build_time" => BUILD_TIME.iso8601,
```

Comment:
You should *never* be comparing a timestamp directly. It's much safer to parse the timestamps into objects, and then use the Time library to compare them. By parsing them into `Time` objects, you'll get comparisons for free:

```
>> rfc_time = Time.now.rfc2822
=> "Thu, 06 Jul 2017 13:48:12 -0700"
>> iso_time = Time.now.iso8601
=> "2017-07-06T13:48:22-07:00"
>> Time.parse(iso_time) - Time.parse(rfc_time)
=> 10.0
>> Time.parse(iso_time) > Time.parse(rfc_time)
=> true
>> Time.parse(iso_time) < Time.parse(rfc_time)
=> false
```